### PR TITLE
update range-v3

### PR DESCRIPTION
**Pull request by h-2**
It affected more stuff than hoped for.

Still depends on:
  * ~~ericniebler/range-v3#1305 / ericniebler/range-v3#1307~~

Follow-Up cards:
  * reactivate a test after ericniebler/range-v3#1320 is fixed
  * uncomment the code-change in seqan3/std/concepts after ericniebler/range-v3#1322 is fixed
  * redesign sequence_builder

### CODE REVIEW

File: include/seqan3/range/views/to.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::views::to.
+ */
+
+#pragma once
+
+#include <range/v3/view/zip.hpp>
```
Comment (rrahn) [inline include/seqan3/range/views/to.hpp]: should this not be `#include <range/v3/range/conversion.hpp>`

File: include/seqan3/range/views/to.hpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::views
+{
+
+//TODO reimplement me
+using ::ranges::to;
```
Comment (rrahn) [inline include/seqan3/range/views/to.hpp]: ranges::to is not in the view namespace of the ranges library. There must be a rational for this. Why are you breaking it?

File: include/seqan3/range/views/join.hpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Provides seqan3::views::zip.
```
Comment (rrahn) [inline include/seqan3/range/views/join.hpp]: does not seem fitting

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp [hunk 4]
```diff
@@ -174,8 +176,27 @@ class aligned_sequence_builder
 
         matrix_coordinate path_begin = trace_it.coordinate();
 
-        fst_aligned_t fst_aligned{fst_rng | views::slice(path_begin.col, path_end.col)};
-        sec_aligned_t sec_aligned{sec_rng | views::slice(path_begin.row, path_end.row)};
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: Oh boy, I see the problem I have missed before. A vector cannot be created from a range. And the gap decorator cannot  be created from a pair of iterators 🤦‍♂ . Well, I need to think about this. I had introduced a type trait class in order to allow the usage of it in a more generic context. This however, defeats the purpose of it. I will rethink the design. But it feels more like a policy based design then.

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp [hunk 5]
```diff
@@ -20,8 +20,10 @@
 #include <seqan3/alignment/matrix/trace_directions.hpp>
 #include <seqan3/alphabet/gap/gapped.hpp>
 #include <seqan3/range/decorator/gap_decorator.hpp>
+#include <seqan3/range/views/convert.hpp>
```
Comment (rrahn) [inline include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: not used?

File: include/seqan3/range/views/zip.hpp [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+#pragma once
+
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/ranges>
+
+namespace ranges::cpp20::views //TODO rename me to seqan3::views
```
Comment (rrahn) [inline include/seqan3/range/views/zip.hpp]: i can't follow?

File: include/seqan3/range/views/to.hpp (hunk 2 above)
Comment (h-2) [reply include/seqan3/range/views/to.hpp]: Well, this is a discussion point I still had. The ranges library places it in `std::ranges::to`, but we don't have a ranges namespace and naming it `seqan3::to` seems too generic for me.

Also we have the notion of "views", "source-only-views", "sink-only-views" (`to` falls into the last category); the range library calls these "view adaptors", "view factories" and has no name for the last. 

Since `to` can only be applied to views, I would put it in `seqan3::views::` but we can discuss that tomorrow.

File: include/seqan3/range/views/zip.hpp (hunk 6 above)
Comment (h-2) [reply include/seqan3/range/views/zip.hpp]: That's one of the TODOs, still need to do a search on replace for this one.

File: include/seqan3/range/views/to.hpp (hunk 2 above)
Comment (rrahn) [reply include/seqan3/range/views/to.hpp]: Mhmm, I see. I think about it but unfortunately, I won't be there tomorrow. I had to take spontaneous vacation. (also wrote it into the gitter channel, sorry for the circumstances). I think it is a little unintuitive what it does then, also since we have views::convert which converts the range element wise. Do we have a sink-only view? Apparently, by our current definition this would not be a sink-only view since it returns a range 😏. Anyway, I find something like `to_container` or `materialize` more intuitive for the user as it more clearly states the intention.

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp (hunk 4 above)
Comment (smehringer) [reply include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: There is the `assign_unaligned` function that lets you assign a range to a vector or gap_decorator with a common interface

File: doc/tutorial/sequence_file/sequence_file_solution3.cpp [hunk 7]
```diff
@@ -65,10 +65,15 @@ int main()
...
 
-    // But you can also do this :)
-    std::vector<std::string> ids = std::move(fin | length_filter | std::views::take(2) | views::get<field::ID>);
+    // But you can also do this to retrieve all IDs into a vector:
+    std::vector<std::string> ids = fin
+                                 | length_filter                                // apply length filter
+                                 | std::views::take(2)                          // take first two records
+                                 | views::get<field::ID>                        // select only ID from record
+                                 | views::convert<std::string &&>               // mark ID to be moved out of record
+                                 | views::to<std::vector<std::string>>;  // convert to container
```
Comment (eseiler) [inline doc/tutorial/sequence_file/sequence_file_solution3.cpp]:
```suggestion
                                 | views::to<std::vector<std::string>>;         // convert to container
```

Comment (eseiler) [reply doc/tutorial/sequence_file/sequence_file_solution3.cpp]: Moving wouldn't work because of `views::to<std::vector<std::string>>`?

File: include/seqan3/alphabet/composite/alphabet_tuple_base.hpp [hunk 8]
```diff
@@ -509,20 +509,20 @@ class alphabet_tuple_base :
     template <typename indirect_component_type>
     constexpr bool operator==(indirect_component_type const rhs) const noexcept
     //!\cond
-        requires detail::one_component_is<alphabet_tuple_base, derived_type, detail::weakly_equality_comparable_with, indirect_component_type>
+        requires detail::one_component_is<alphabet_tuple_base, derived_type, detail::weakly_equality_comparable_with_, indirect_component_type>
```
Comment (eseiler) [inline include/seqan3/alphabet/composite/alphabet_tuple_base.hpp]: Line lengths in this file

File: include/seqan3/alphabet/detail/alphabet_proxy.hpp [hunk 9]
```diff
@@ -301,7 +301,7 @@ class alphabet_proxy : public alphabet_base<derived_type,
     //!\brief Allow (in-)equality comparison with types that the emulated type is comparable with.
     template <typename t>
     friend constexpr auto operator==(derived_type const lhs, t const rhs) noexcept
-        -> std::enable_if_t<!std::same_as<derived_type, t> && std::detail::weakly_equality_comparable_with<alphabet_type, t>,
+        -> std::enable_if_t<!std::same_as<derived_type, t> && seqan3::detail::weakly_equality_comparable_with<alphabet_type, t>,
```
Comment (eseiler) [inline include/seqan3/alphabet/detail/alphabet_proxy.hpp]: Line lengths in this file

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 10]
```diff
@@ -188,7 +189,7 @@ class sequence_file_input_format_REMOVEME<format_fasta>
                 std::ranges::copy(stream_view | std::views::drop_while(is_id || is_blank)        // skip leading >
                                               | views::take_until_or_throw(is_cntrl || is_blank) // read ID until delimiter…
                                               | views::char_to<value_type_t<id_type>>,
-                                  std::back_inserter(id));                                      // … ^A is old delimiter
+                                  std::ranges::back_inserter(id));                                      // … ^A is old delimiter
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fasta.hpp]:
```suggestion
                                  std::ranges::back_inserter(id));                              // … ^A is old delimiter
```

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 11]
```diff
@@ -280,7 +281,7 @@ class sequence_file_input_format_REMOVEME<format_fasta>
                                                 return c;
                                             })                                           // enforce legal alphabet
                                           | views::char_to<value_type_t<seq_type>>,       // convert to actual target alphabet
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fasta.hpp]:
```suggestion
                                           | views::char_to<value_type_t<seq_type>>,      // convert to actual target alphabet
```

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 12]
```diff
@@ -398,7 +399,7 @@ class sequence_file_output_format_REMOVEME<format_fasta>
             //TODO: combining chunk and join is substantially faster than views::interleave (2.5x), why?
             std::ranges::copy(seq | views::to_char
                                   | ranges::view::chunk(options.fasta_letters_per_line)
-                                  | std::views::join(options.add_carriage_return
+                                  | views::join(options.add_carriage_return
                                                     ? std::string_view{"\r\n"}
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fasta.hpp]:
```suggestion
                                               ? std::string_view{"\r\n"}
```

File: include/seqan3/io/sequence_file/format_fasta.hpp [hunk 13]
```diff
@@ -398,7 +399,7 @@ class sequence_file_output_format_REMOVEME<format_fasta>
...
                                                     : std::string_view{"\n"}),
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/format_fasta.hpp]:
```suggestion
                                               : std::string_view{"\n"}),
```

File: include/seqan3/std/concepts [hunk 14]
```diff
@@ -50,14 +51,62 @@
...
 
-namespace std
-{
-
 /*!\defgroup concepts concepts
  * \ingroup std
  * \brief The \<concepts\> header from C++20's standard library.
  */
 
+#if 0
```
Comment (eseiler) [inline include/seqan3/std/concepts]: ?

File: doc/tutorial/sequence_file/sequence_file_solution3.cpp (hunk 7 above)
Comment (h-2) [reply doc/tutorial/sequence_file/sequence_file_solution3.cpp]: move what?

Comment (eseiler) [reply doc/tutorial/sequence_file/sequence_file_solution3.cpp]: Like before, `std::move( bla | bla | bla)`

**Comment by h-2** [comment]
We are now based on the remote branch that includes the fixes. One local test deactivated (tests behaviour from range-v3).

File: doc/tutorial/sequence_file/sequence_file_solution3.cpp (hunk 7 above)
Comment (eseiler) [reply doc/tutorial/sequence_file/sequence_file_solution3.cpp]: You missed the suggestion 😁

**Comment by h-2** [comment]
I am still waiting for upstream to merge the branch...

File: include/seqan3/io/sequence_file/output.hpp [hunk 15]
```diff
@@ -592,12 +593,16 @@ class sequence_file_output
     template <typename ...arg_types>
     sequence_file_output & operator=(std::tuple<arg_types...> const & t)
     {
+        // remove the following after range-v3 is updated to 1.0
```
Comment (eseiler) [inline include/seqan3/io/sequence_file/output.hpp]: I trust there is a card for this?

File: include/seqan3/range/views/join.hpp [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <range/v3/view/join.hpp>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::views
+{
+
+//TODO reimplement me and rename to join_with
```
Comment (eseiler) [inline include/seqan3/range/views/join.hpp]: Are there cards?
Might as well do \todo

File: include/seqan3/std/concepts [hunk 17]
```diff
@@ -51,14 +52,62 @@
...
```
Comment (eseiler) [inline include/seqan3/std/concepts]: What was the rationale?
I vaguely remember that there was a reason...

File: test/unit/range/views/view_interleave_test.cpp [hunk 18]
```diff
@@ -87,6 +89,7 @@ TEST(view_interleave, concepts)
     EXPECT_FALSE((std::ranges::output_range<decltype(v2), dna4>));
 }
 
+#if 0 // blocked by https://github.com/ericniebler/range-v3/issues/1320
```
Comment (eseiler) [inline test/unit/range/views/view_interleave_test.cpp]: There's a card?
Was there a similar reason for the other `#if 0`

File: test/unit/range/views/view_persist_test.cpp [hunk 19]
```diff
@@ -30,60 +31,60 @@ TEST(view_persist, delegate_to_view_all)
...
-    std::string v3b = vec | std::views::reverse | views::persist | ranges::view::unique | std::ranges::to<std::string>;
+    EXPECT_EQ("fo", v3 | views::to<std::string>);
+    std::string v3b = vec | std::views::reverse | views::persist | ranges::view::unique | views::to<std::string>;
     EXPECT_EQ("of", v3b);
 
-    // store combined
-    auto a1 = views::persist | ranges::view::unique;
-    auto v5 = vec | a1;
-    EXPECT_EQ("fo", std::string{v5});
+    // store combined  TODO don't merge this:
```
Comment (eseiler) [inline test/unit/range/views/view_persist_test.cpp]: 👀

File: test/unit/range/views/view_persist_test.cpp [hunk 20]
```diff
@@ -30,60 +31,60 @@ TEST(view_persist, delegate_to_view_all)
...
-    // combinability
-    auto v3 = std::string{"foo"} | views::persist | ranges::view::unique;
-    EXPECT_EQ("fo", std::string(v3));
-    std::string v3b = std::string{"foo"}
-                    | views::persist
-                    | std::views::filter(is_char<'o'>)
-                    | ranges::view::unique
-                    | std::ranges::to<std::string>;
-    EXPECT_EQ("o", v3b);
+    // combinability TODO don't merge this:
```
Comment (eseiler) [inline test/unit/range/views/view_persist_test.cpp]: 👀

File: include/seqan3/range/views/join.hpp (hunk 16 above)
Comment (h-2) [reply include/seqan3/range/views/join.hpp]: \todo is only parsed as part of doxygen comments..

File: include/seqan3/io/sequence_file/output.hpp (hunk 15 above)
Comment (h-2) [reply include/seqan3/io/sequence_file/output.hpp]: There will be :tm:

File: include/seqan3/std/concepts (hunk 17 above)
Comment (h-2) [reply include/seqan3/std/concepts]: The boolean concept is currently broken, see the issues linked above.

File: test/unit/range/views/view_interleave_test.cpp (hunk 18 above)
Comment (h-2) [reply test/unit/range/views/view_interleave_test.cpp]: There will be, see above.

File: include/seqan3/range/views/join.hpp (hunk 16 above)
Comment (eseiler) [reply include/seqan3/range/views/join.hpp]: Why no do it as comment then.
Doxygen might complain anyway about it being not documented...

File: include/seqan3/std/concepts (hunk 17 above)
Comment (eseiler) [reply include/seqan3/std/concepts]: Ok so this should link to the issue and should be activated once the issue is fixed?

File: include/seqan3/range/views/join.hpp (hunk 16 above)
Comment (h-2) [reply include/seqan3/range/views/join.hpp]: Then I have to cond it to not make it appear in the user documentation, I really think having a card is the way to go and not use doxygen for this.

**Comment by h-2** [comment]
I have added cards for everything. Note that we need to go through the library and look for range-v3 includes anyway. This is not resolved by this PR.

I already had to do some nasty rebasing, so it would be great if we can get this in soon. @eseiler if you are happy, please assign @smehringer with the final review.

File: doc/tutorial/sequence_file/sequence_file_solution3.cpp (hunk 7 above)
Comment (eseiler) [reply doc/tutorial/sequence_file/sequence_file_solution3.cpp]: It's back!

**Comment by h-2** [comment]
> One change was reverted, but you can do it with the next push

That's what I meant, the rebases are becoming more and more annoying :sob:

**Comment by eseiler** [comment]
> I have no clue why the warning is triggered in the header test. I cannot reproduce it locally. The only workaround I see currently is deactivating the deprecated warning for the entire `sequence_file/{input.hpp|output.hpp}`
I had a similar problem with the fm_index, where the deprecations wouldn't be ignored unless I set the pragma before including anything.
> 
> As already documented for the other bits: this will go away once ranges removes the implicit cast.

File: include/seqan3/range/container/concatenated_sequences.hpp [hunk 21]
```diff
@@ -1329,7 +1326,7 @@ class concatenated_sequences
     //!\brief Checks whether `*this` is less than or equal to `rhs`.
     constexpr bool operator<=(concatenated_sequences const & rhs) const noexcept
     {
-        return raw_data() <= rhs.data();
+        return raw_data() <= rhs.raw_data();
```
Comment (smehringer) [inline include/seqan3/range/container/concatenated_sequences.hpp]: this seems unrelated to the ranges update?
Whats happening?

File: include/seqan3/range/views/interleave.hpp [hunk 22]
```diff
@@ -294,8 +295,8 @@ struct interleave_fn
         }
         else
         {
-            return ranges::view::chunk(std::forward<urng_t>(urange), static_cast<size_t>(size))
-                 | std::views::join(std::forward<inserted_rng_t>(i));
+            return std::forward<urng_t>(urange) | ranges::views::chunk(static_cast<size_t>(size))
+                                                | ranges::views::join(std::forward<inserted_rng_t>(i));
```
Comment (smehringer) [inline include/seqan3/range/views/interleave.hpp]: ~~ranges::~~ views::join

File: include/seqan3/range/views/join.hpp (hunk 3 above)
Comment (smehringer) [reply include/seqan3/range/views/join.hpp]: not resolved

File: include/seqan3/range/views/join.hpp [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+#pragma once
+
+#include <range/v3/view/join.hpp>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/ranges>
+
+namespace seqan3::views //TODO rename me to seqan3::views
```
Comment (smehringer) [inline include/seqan3/range/views/join.hpp]: remove todo

File: include/seqan3/range/views/to.hpp (hunk 1 above)
Comment (smehringer) [reply include/seqan3/range/views/to.hpp]: not resolved?

File: include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp [hunk 24]
```diff
@@ -477,7 +478,7 @@ void stream_alignment(debug_stream_type<char_t> & stream, alignment_t const & al
         {
             // write alignment bars
             stream << '\n' << std::setw(8) << "";
-            std::ranges::for_each(ranges::zip_view(previous_seq, aligned_seq) | views::slice(begin_pos, end_pos),
+            std::ranges::for_each(views::zip(previous_seq, aligned_seq) | views::slice(begin_pos, end_pos),
                              [&stream] (auto && ch) { stream << (get<0>(ch) == get<1>(ch) ? '|' : ' '); });
```
Comment (smehringer) [inline include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp]: indentation

File: include/seqan3/core/type_traits/iterator.hpp [hunk 25]
```diff
@@ -127,34 +127,6 @@ struct size_type<it_t>
  * \tparam it_t The type to operate on.
  * \see seqan3::iterator_tag_t
  *
- * <table>
- *   <tr>
- *     <th>Modelled concept</th>
```
Comment (smehringer) [inline include/seqan3/core/type_traits/iterator.hpp]: why did you remove this?

File: include/seqan3/search/fm_index/fm_index.hpp [hunk 26]
```diff
@@ -18,6 +18,7 @@
 #include <seqan3/std/filesystem>
 #include <seqan3/range/shortcuts.hpp>
 #include <seqan3/range/views/to_rank.hpp>
+#include <seqan3/range/views/to.hpp>
```
Comment (smehringer) [inline include/seqan3/search/fm_index/fm_index.hpp]: import `seqan3/views/join` too

File: include/seqan3/std/algorithm [hunk 27]
```diff
@@ -37,80 +39,4 @@
...
 
-#include <seqan3/core/platform.hpp>
-
-namespace std::ranges
-{
-
-/*!\typedef std::ranges::all_of
- * \brief Alias for ranges::all_of. Checks if unary predicate `p` returns `true` for all elements in the range.
- */
-using SEQAN3_DOXYGEN_ONLY(all_of =) ::ranges::all_of;
```
Comment (smehringer) [inline include/seqan3/std/algorithm]: I don't understand this change :see_no_evil: 
Where do you alias the range algorithms now?
Why are all the range includes still in there if not used?

File: include/seqan3/std/concepts [hunk 28]
```diff
@@ -22,7 +22,8 @@
 #include <functional>
 #include <type_traits>
 
-#include <range/v3/utility/common_type.hpp>
+#include <concepts/concepts.hpp>
```
Comment (smehringer) [inline include/seqan3/std/concepts]: huh? where does this come from?

File: test/performance/alignment/global_affine_alignment_parallel_benchmark.cpp [hunk 29]
```diff
@@ -17,6 +17,7 @@
 #include <seqan3/alignment/pairwise/align_pairwise.hpp>
 #include <seqan3/alphabet/aminoacid/aa20.hpp>
 #include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/range/views/to.hpp>
```
Comment (smehringer) [inline test/performance/alignment/global_affine_alignment_parallel_benchmark.cpp]: also include `seqan3/views/zip`

**Comment by smehringer** [comment]
In general you commit history is a bit chaotic I feel.
You introduce the zip header in one commit but change the naming in another. also "more fixes" doesn't really help in a commit message :wink: Maybe just squash all of them except the async fix

File: include/seqan3/range/views/to.hpp (hunk 1 above)
Comment (h-2) [reply include/seqan3/range/views/to.hpp]: Oh no, then more got lost in the rebases..

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp [hunk 30]
```diff
@@ -94,15 +96,15 @@ class aligned_sequence_builder
...
-    static_assert(std::constructible_from<typename traits_t::fst_aligned_t,
-                                     decltype(std::declval<fst_rng_t>() | views::slice(0, 1))>,
-                  "The given sequence type cannot be transformed to a aligned_sequence. It needs to be a views::slice "
-                  "over the original type.");
-
-    static_assert(std::constructible_from<typename traits_t::sec_aligned_t,
-                                     decltype(std::declval<sec_rng_t>() | views::slice(0, 1))>,
-                  "The given sequence type cannot be transformed to a aligned_sequence. It needs to be a views::slice "
-                  "over the original type.");
+//     static_assert(std::constructible_from<typename traits_t::fst_aligned_t,
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: Because it's not constructible anymore.

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp [hunk 31]
```diff
@@ -174,8 +176,27 @@ class aligned_sequence_builder
 
         matrix_coordinate path_begin = trace_it.coordinate();
 
-        fst_aligned_t fst_aligned{fst_rng | views::slice(path_begin.col, path_end.col)};
-        sec_aligned_t sec_aligned{sec_rng | views::slice(path_begin.row, path_end.row)};
+        fst_aligned_t fst_aligned;
+        if constexpr (detail::is_type_specialisation_of_v<fst_aligned_t, std::vector>)
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: It's not about assigning unaligned. It's about assigning aligned, butsometimes vector and sometimes adaptor. In any case the whole design is broken and @rrahn agreed to fix it, but not as part of this PR.

File: include/seqan3/alphabet/composite/detail.hpp [hunk 32]
```diff
@@ -132,11 +132,11 @@ struct assignable_from
  * Returns an std::true_type if the `type` is weakly equality comparable to `T`.
  */
 template <typename T>
-struct weakly_equality_comparable_with
+struct weakly_equality_comparable_with_
```
[the remark this answers is not in the document]
Comment (h-2) [reply include/seqan3/alphabet/composite/detail.hpp]: `_helper` also has no meaning at all :wink: 

No seriously, @joshuak94 is currently removing the meta stuff and I have created a card already to simplify all this code in the composites: https://github.com/seqan/seqan3/projects/1#card-27117113

File: include/seqan3/range/container/concatenated_sequences.hpp (hunk 21 above)
Comment (h-2) [reply include/seqan3/range/container/concatenated_sequences.hpp]: Obviously @eseiler missed this if you look at the surrounding code. I am not sure why, but after my changes, this path got triggered and I got a deprecated warning. It needs be fixed in any case.

File: include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp (hunk 24 above)
Comment (h-2) [reply include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp]: Of what excactly? The lambda expression? I can indent it, but it's not something I touched in this PR.

File: include/seqan3/core/type_traits/iterator.hpp (hunk 25 above)
Comment (h-2) [reply include/seqan3/core/type_traits/iterator.hpp]: The links were all broken, because there is no documentation for the concepts any more. It will appear once there is on cppreference, but making "conditional" links doesn't work inside the html table (why no markdown!?). 
I tried changing it to markdown, but then it looked different. Then I realised the whole data structure is very broken and that we should remove it. I made a card for it https://github.com/seqan/seqan3/projects/3#card-27178357

Comment (h-2) [reply include/seqan3/core/type_traits/iterator.hpp]: (its also detail)

File: include/seqan3/std/algorithm (hunk 27 above)
Comment (h-2) [reply include/seqan3/std/algorithm]: > Where do you alias the range algorithms now?

The entire namespace is imported.

> Why are all the range includes still in there if not used?

They are used :smile:

File: include/seqan3/std/concepts (hunk 28 above)
Comment (h-2) [reply include/seqan3/std/concepts]: The ranges library, there is a code comment on this a few lines below:
```cpp
// from concept/concepts.hpp (also range-v3)
```

**Comment by h-2** [comment]
> In general you commit history is a bit chaotic I feel.

Yes, because other people keep breaking stuff :crying_cat_face: 

> Maybe just squash all of them except the async fix

Ok.

>>> The only workaround I see currently is deactivating the deprecated warning for the entire 

You didn't comment on this point of mine. I guess I will just do it like that now,

>> Maybe you could also change it here:

@eseiler I think it's more likely we forget to change it there later on, or not? 

>> When will this be?

I suspect with range-v3-1.0 which I expect after the ISO February meeting when C++20 is published.

**Comment by eseiler** [comment]
> > > Maybe you could also change it here:
> 
> @eseiler I think it's more likely we forget to change it there later on, or not?

We changed it there for other stuff (like the fm_index when the sdsl wasn't updated to be header-test-safe). 
Either way is fine, just add a card for it :)
> 
> > > When will this be?
> 
> I suspect with range-v3-1.0 which I expect after the ISO February meeting when C++20 is published.

Ok.

**Comment by h-2** [comment]
If my last push doesn't fix travis I would propose deactivating the warning for the header build or moving the header build to gcc8. After all any use of deprecated things will be detected by the unit tests and snippets.

**Comment by h-2** [comment]
If you keep on breaking this with other PRs, someone else will need to take over this PR :roll_eyes:

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp (hunk 31 above)
[the remark this answers is not in the document]
Comment (smehringer) [reply include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: ok

File: include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp (hunk 30 above)
[the remark this answers is not in the document]
Comment (smehringer) [reply include/seqan3/alignment/matrix/detail/aligned_sequence_builder.hpp]: I guess this also will be fixed in the refactorisation of rene

**Comment by rrahn** [comment]
> If you keep on breaking this with other PRs, someone else will need to take over this PR 

Should we stop merging other PRs until this is done now? If so can you please give a heads up on gitter then? Or did I  missed the notification there?

**Comment by h-2** [comment]
> Should we stop merging other PRs until this is done now? If so can you please give a heads up on gitter then? Or did I missed the notification there?

Well, I thought we had discussed that this PR has relatively high priority, because many other things depend on it. But maybe I should have communicated this more clearly, no offence to anyone intended.
However, I definitely won't be working on SeqAn-related things from tomorrow till sometime next week.

**Comment by rrahn** [comment]
> Well, I thought we had discussed that this PR has relatively high priority, because many other things depend on it. But maybe I should have communicated this more clearly, no offence to anyone intended.
However, I definitely won't be working on SeqAn-related things from tomorrow till sometime next week.

Yes, but on Monday it sounded like it still depends on stuff, which should be there soon. But for me this was not a concrete hint for not merging other stuff yet. But maybe I was also wrong sorry. But anyway, we wait now (to long) for the CI to get done and then merge it hopefully.

**Comment by smehringer** [comment]
@h-2 some tests are failing :(

**Comment by h-2** [comment]
I have now changed the header tests to use GCC8, let's see if that helps. (I also addressed the other issues)

**Comment by eseiler** [comment]
> I have now changed the header tests to use GCC8, let's see if that helps. (I also addressed the other issues)

Still breaks.. and the search module also gets errors.
maybe just exclude the files from the header test for now and I'll look into it as a follow up issue.

**Comment by h-2** [comment]
I have reverted header tests to gcc7 but deactivated the specific warning on travis (I hope).

In general the setup of travis seems quite broken if a multitude of things happen that no one can reproduce.

**Comment by eseiler** [comment]
> I have reverted header tests to gcc7 but deactivated the specific warning on travis (I hope).
> 
> In general the setup of travis seems quite broken if a multitude of things happen that no one can reproduce.

I'll run the travis ci locally and try to figure out what's so special about the environment...until then we just ignore the warnings.

**Comment by h-2** [comment]
I don't know what's going on with Jenkins, there is no clickable link. If I go to the page (and it loads after multiple minutes) it tells me "Recursively delete the current directory from the workspace" but there is also no log...

I will look again at this later, but have to work on other things now. Feel free to remove the last commit and/or push more commits on this branch.

**Comment by h-2** [comment]
Merging this as all CI passes. Sorry for blocking the merge queue.
